### PR TITLE
implemented delete to file and folder

### LINKED ISSUE #101: Delete file
Delete the file's metadata from the DB.
Not including Folders for now.

### PR DESCRIPTION
**Pull request by Shahar-Y**
delete works, checked kubernetes 
closes #101, #102
Signed-off-by: shahar-Y <shya95@gmail.com>

### CODE REVIEW

File: src/file/file.model.ts [hunk 1]
```diff
@@ -81,7 +77,7 @@ fileSchema.virtual('fullExtension')
 
 fileSchema.pre('save', async function (next: NextFunction) {
   const existingFile = await fileModel.findOne({ name: (<any>this).name, parent: (<any>this).parent, ownerID: (<any>this).ownerID });
-  if (existingFile && !existingFile.deleted) {
+  if (existingFile) {
```
Comment (omrishtam) [inline src/file/file.model.ts]: If you don't have `deleted` field anymore then you should make a unique index of `name`, `parent` and `ownerID`.

File: src/file/file.service.ts [hunk 2]
```diff
@@ -73,8 +73,8 @@ export class FileService {
   /**
    * If receiving a file, deletes the file.
    * If receiving a folder:
-   * Recursively delete all files and sub-folders in the given folder,
-   * only then delete the folder.
+   * *Recursively delete all files and sub-folders in the given folder,
```
Comment (omrishtam) [inline src/file/file.service.ts]: Remove dup `*`.

File: src/file/file.model.ts (hunk 1 above)
Comment (Shahar-Y) [reply src/file/file.model.ts]: done

File: src/file/file.service.ts (hunk 2 above)
Comment (Shahar-Y) [reply src/file/file.service.ts]: done

File: src/file/file.model.ts (hunk 1 above)
Comment (Shahar-Y) [reply src/file/file.model.ts]: @omrishtam any comments or is it resolved?

Comment (omrishtam) [reply src/file/file.model.ts]: I meant that you should remove the `pre` hook, there's no need for it if you can just make an index.

Comment (Shahar-Y) [reply src/file/file.model.ts]: hmm.. you are right, but I still want to give a good error response (which I started to implement on the hotfix branch I created today). So I guess I'll delete the pre on this branch and handle it on the other.
TODO on sunday.

Comment (Shahar-Y) [reply src/file/file.model.ts]: removed

**Pull request merged**